Fix build with curl 8.16.0

Downloaded from
https://gitlab.alpinelinux.org/alpine/aports/-/commit/106a02411488e3f71b9f2085a4a857e4be27b9ea

Upstream: https://github.com/xbmc/xbmc/pull/27224

The upstream patch is aimed at the upcoming release of Kodi 22.0 which
saw many changes to CurlFile.cpp so we use the fix from Alpine.

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>

--- a/xbmc/filesystem/CurlFile.cpp
+++ b/xbmc/filesystem/CurlFile.cpp
@@ -44,7 +44,7 @@ using namespace std::chrono_literals;
 
 #define FITS_INT(a) (((a) <= INT_MAX) && ((a) >= INT_MIN))
 
-curl_proxytype proxyType2CUrlProxyType[] = {
+long proxyType2CUrlProxyType[] = {
     CURLPROXY_HTTP,   CURLPROXY_SOCKS4,          CURLPROXY_SOCKS4A,
     CURLPROXY_SOCKS5, CURLPROXY_SOCKS5_HOSTNAME, CURLPROXY_HTTPS,
 };

